parquet-generator/src/main/java/org/apache/parquet/filter2/IncrementallyUpdatedFilterPredicateGenerator.java

```diff
@@ -299,22 +304,23 @@ private void addInNotInCase(TypeInfo info, boolean isEq) throws IOException {
         + "\n"
         + "          @Override\n"
         + "          public void update("
-        + info.primitiveName + " value) {\n" + "            boolean set = false;\n");
+        + info.primitiveName + " value) {\n");
 
+    if (expectMultipleResults) {
+      add("            if (isKnown()) return;\n");
+    }
     add("            for (" + info.primitiveName + " i : target) {\n");
 
     add("              if(" + compareEquality("value", "i", isEq) + ") {\n");
 
-    add("                 setResult(true);\n");
-
-    add("                 set = true;\n");
-
-    add("                 break;\n");
+    add("                 setResult(true);\n                 return;\n");
```

Comment:
I simplified the existing ValueInspector logic a bit here. Previously it was:

```java
  @Override
  public void update(int value) {
	boolean set = false;
	for (int i : target) {
	  if(comparator.compare(value, i) != 0) {
		 setResult(true);
		 set = true;
		 break;
	   }
	 }
	 if (!set) setResult(false);
   }
 };
}
```

Now it's:
```java
@Override
public void update(int value) {
for (int i : target) {
  if(comparator.compare(value, i) == 0 ) {
	 setResult(true);
	 return;
   }
 }
 setResult(false);
}
```